src/Geometry/Primitives/NonCSG/ConvexPolygon.jl

```diff
@@ -13,50 +14,62 @@ The rotation of the polygon around its normal is defined by `rotationvec`.
 
     # for efficency
-    _local_frame_inv::Transform{T}                                 # cache the inverse matrix to avoid computing it for every intersection test
-    _poly2d::GeometricalPredicates.Polygon2D{GeometricalPredicates.Point2D}
+    _local_frame_inv::Transform{T}                                  # cache the inverse matrix to avoid computing it for every intersection test
+    _local_lines::SVector{N, SVector{3, SVector{2, T}}}             # defines the edges + a third point representing the slopes in order to save some calculationsduring ray checking
+    _length::Int64
+    # _poly2d::GeometricalPredicates.Polygon2D{GeometricalPredicates.Point2D}
 
     function ConvexPolygon(
             local_frame::Transform{T},
-            local_polygon_points::Vector{SVector{2, T}}, 
+            local_polygon_points::SVector{N, SVector{2, T}}, 
             interface::NullOrFresnel{T} = NullInterface(T)
-        ) where {T<:Real}
+        ) where {N, T<:Real}
 
         # need at least 3 points to define apolygon
         @assert length(local_polygon_points) > 3         
 
         local_center = Statistics.mean(local_polygon_points)
         world_center = local2world(local_frame) * Vec3(local_center[1], local_center[2], zero(T))
 
-        poly_points = [GeometricalPredicates.Point2D(p[1], p[2]) for p in local_polygon_points]
-        poly = GeometricalPredicates.Polygon2D(poly_points...)
+        
+        # poly_points = [GeometricalPredicates.Point2D(p[1], p[2]) for p in local_polygon_points]
+        # poly = GeometricalPredicates.Polygon2D(poly_points...)
+        pts = local_polygon_points
+        local_lines = SVector(
+            [SVector(
+                pts[i],                                             # A
+                pts[mod(i, length(pts))+1],                         # B
+                                                                    # bx = Bx - Ax,  by = By - Ay
+                SVector(pts[mod(i, length(pts))+1][1] - pts[i][1], pts[mod(i, length(pts))+1][2] - pts[i][2]))
+                for i in 1:length(pts)]...
+        )
 
         plane = Plane(forward(local_frame), world_center, interface = interface)
-        new{T}(plane, local_frame, local_polygon_points, inv(local_frame), poly)
+        new{N, T}(plane, local_frame, local_polygon_points, inv(local_frame), local_lines, length(local_lines))
     end
 end
 export ConvexPolygon
 
 # Base.show(io::IO, poly::ConvexPolygon{T}) where {T<:Real} = print(io, "ConvexPolygon{$T}($(centroid(hex)), $(normal(hex)), $(hex.side_length), $(interface(hex)))")
-centroid(poly::ConvexPolygon{T}) where {T<:Real} = poly.plane.pointonplane
-interface(poly::ConvexPolygon{T}) where {T<:Real} = interface(poly.plane)
-normal(poly::ConvexPolygon{T}) where {T<:Real} = normal(poly.plane)
-normal(poly::ConvexPolygon{T}, ::T, ::T) where {T<:Real} = normal(poly)
-
+centroid(poly::ConvexPolygon{N, T}) where {N, T<:Real} = poly.plane.pointonplane
```

Comment:
Since N and T are not being used on the right hand side of the = you shouldn't need them. Should still generate efficient code if you remove them.